modules/tinymce/src/themes/silver/test/ts/browser/throbber/ThrobberFocusTest.ts

```diff
@@ -0,0 +1,194 @@
+    await pAssertThrobberHidden();
+    await pAssertFocus();
+  };
+
+  it('TINY-7373: should focus throbber when enabled and focus editor when disabled', async () => {
+    const editor = hook.editor();
+    await pEnableThrobber(editor, pAssertThrobberFocus);
+    assert.isFalse(editor.hasFocus());
+    await pDisableThrobber(editor, pAssertEditorFocus(editor));
+    assert.isTrue(editor.hasFocus());
+  });
+
+  it('TINY-7373: should not steal focus if editor is not focused when throbber is enabled', async () => {
+    const editor = hook.editor();
+    const pAssertInputFocus = () => pAssertFocus('Focus on input', '#tempInput');
+
+    FocusTools.setFocus(SugarBody.body(), '#tempInput');
+    await pEnableThrobber(editor, pAssertInputFocus);
+
+    await pDisableThrobber(editor, pAssertInputFocus);
+    assert.isFalse(editor.hasFocus());
+  });
+
+  it('TINY-7373: should take focus if editor.focus() is called', async () => {
+    const editor = hook.editor();
+
+    editor.focus();
+    await pEnableThrobber(editor, pAssertThrobberFocus);
+    editor.focus();
+    await pAssertThrobberFocus();
+    assert.isFalse(editor.hasFocus());
+
+    await pDisableThrobber(editor, pAssertEditorFocus(editor));
+    assert.isTrue(editor.hasFocus());
+  });
+
+  it('TINY-7373: should not take focus if editor.focus(true) is called', async () => {
+    const editor = hook.editor();
+    const pAssertInputFocus = () => pAssertFocus('Focus on input', '#tempInput');
+
+    FocusTools.setFocus(SugarBody.body(), '#tempInput');
+    await pEnableThrobber(editor, pAssertInputFocus);
+
+    editor.focus(true);
+    await pAssertInputFocus();
+    assert.isFalse(editor.hasFocus());
+    editor.focus();
+    await pAssertThrobberFocus();
+
+    await pDisableThrobber(editor, pAssertEditorFocus(editor));
+    assert.isTrue(editor.hasFocus());
+  });
+
+  it('TINY-7373: should have correct focus transitions when opening and closing dialog', async () => {
+    const editor = hook.editor();
+    const pAssertDialogFocus = () => pAssertFocus('Dialog input has focus', 'input.tox-textfield');
+
+    openDialog(editor);
+    await TinyUiActions.pWaitForDialog(editor);
+    await pAssertDialogFocus();
+
+    // Throbber should not steal focus from the dialog
+    await pEnableThrobber(editor, pAssertDialogFocus);
+
+    // Throbber should get focus when dialog is closed instead of editor
+    TinyUiActions.cancelDialog(editor);
+    await Waiter.pTryUntil('Dialog should close', () => UiFinder.notExists(SugarBody.body(), 'div[role="dialog"]'));
+    await pAssertThrobberFocus();
+
+    // Focus returns to the editor when throbber is closed (this should happen since a dialog will normally focus the editor on close)
+    await pDisableThrobber(editor, pAssertEditorFocus(editor));
+    assert.isTrue(editor.hasFocus());
+  });
+
+  // In theory, this combination shouldn't be possible as the user cannot click any buttons to open dialogs when the throbber is enabled
```

Comment:
This is definitely a viable case, as take for example a notification could be shown (as it is with RTC) while the throbber is active.